src/components/SharedFields/SharedFields.tsx

```diff
@@ -0,0 +1,230 @@
+import React, { FC, useState } from 'react';
+import { Input, Select, DatePicker, Form, Checkbox, Row } from 'antd';
+
+import SharedOptions from './SharedOptions';
+
+const SharedFields: FC = () => {
+  // --- useState to store inputs ---
+  const [userInput, updateUserInput] = useState({
```

Comment:
```suggestion
  const [userInput, setUserInput] = useState({
```

Convention with state hooks is to prepend the function that updates the state with "set". Changing this will also involve updating any functions that call `updateUserInput`